test/test_cuda.py

```diff
@@ -3941,7 +3941,8 @@ def _test_reduce_add_coalesced(self, tensors, buffer_size):
         r_tensors = [comm.reduce_add(t) for t in zip(*dup_tensors)]
         for r, t in zip(r_tensors, tensors):
             self.assertEqualTypeString(r, t)
-            self.assertEqual(r, t * 2)
+            maybe_coalesce = lambda x: x.coalesce() if x.is_sparse else x
+            self.assertEqual(maybe_coalesce(r), t * 2)
```

Comment:
Why the lambda if we only use it once?

```suggestion
            self.assertEqual(r.coalesce() if r.is_sparse else r, t * 2)
```